src/api.js

```diff
@@ -1,14 +1,41 @@
 import { cachedRequest } from './cachedRequest'
 import {
 	buildMwApiUrl, buildCommonsApiUrl, convertUrlToMobile,
-	strip, getDeviceSize, sanitizeHTML, getAnalyticsQueryParam
+	strip, getDeviceSize, sanitizeHTML, getAnalyticsQueryParam,
+	getDir
 } from './utils'
 
-const requestPagePreview = ( lang, title, isTouch, callback, request = cachedRequest ) => {
+const requestMwExtract = ( lang, title, isTouch, callback, request = cachedRequest ) => {
+	const params = {
+		action: 'query',
+		prop: 'extracts|pageimages',
+		exsentences: 4,
+		explaintext: 1,
+		exsectionformat: 'plain',
+		piprop: 'thumbnail',
+		pilimit: 1,
+		titles: title
+	}
+	const url = buildMwApiUrl( lang, params )
+	request( url, ( result ) => {
+		const page = result.query.pages[ Object.keys( result.query.pages )[ 0 ] ]
```

Comment:
We need to filter if there is `missing` key in the response, otherwise the undefined looks super bad

| ![image](https://user-images.githubusercontent.com/2560096/128173558-6a00e06c-b05a-46f0-bd37-06d0566bc62f.png) | ![image](https://user-images.githubusercontent.com/2560096/128173876-ed303df2-34df-40c4-a241-413d659bbf5b.png) |
| --- | --- |